### PR TITLE
Add BulletproofPlus and view tags for v15 hardfork 

### PR DESCRIPTION
**Pull request by Boog900**
Add support for De/serialization of Bulletproof+ and view tags
Add a test for Bulletproof+ and view tags
use view tags to speed up owned output finding

### CODE REVIEW

**Comment by hansieodendaal** [comment]
Merge mining proxy error:
```
[tari_mm_proxy::proxy] ERROR Method "getblocktemplate" failed handling request: MergeMiningError { source: ValidationError("blocktemplate blob invalid") }
```

File: src/util/ringct.rs [hunk 1]
```diff
@@ -458,6 +460,33 @@ pub struct Bulletproof {
...
+// ====================================================================
+/// BulletproofPlus format.
+#[derive(Debug, Clone)]
+#[allow(non_snake_case)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "serde", serde(crate = "serde_crate"))]
+pub struct BulletproofPlus {
+    /// A value.
+    pub A: Key,
+    /// S value.
```
Comment (busyboredom) [inline src/util/ringct.rs:472]: I think this comment should say A1.

**Comment by busyboredom** [comment]
Is there anything else that needs to be done for this or is it good to go? I'm happy to help if there's anything else needed to be compatible with v15.

**Comment by Boog900** [comment]
I think it's good to go but I can change anything if needed 

@h4sh3d

**Review by h4sh3d — APPROVED** [review]
Thanks a lot for the work! I think we can release `0.18.0` with this one

**Pull request merged**